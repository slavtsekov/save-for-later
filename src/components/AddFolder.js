import React from 'react';
import { connect } from 'react-redux';
import { addFolder } from '../actions/folders';

export class AddFolder extends React.Component {
    state = {
        error: undefined
    };

    addFolder = (e) => {
        e.preventDefault();

        const folderName = e.target.elements.folder.value.trim();
        if (!folderName) {
            this.setState(() => ({
                error: 'Please provide folder name.'
            }));
        } else {
            this.setState(() => ({
                error: undefined
            }));

            this.props.addFolder(folderName);
            e.target.elements.folder.value = '';
        }
    };

    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form
                    onSubmit={this.addFolder}
                >
                    <input type="text" name="folder" />
                    <button type="submit">Add Folder</button>
                </form>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addFolder: (folder) => { dispatch(addFolder(folder)); }
});

export default connect(undefined, mapDispatchToProps)(AddFolder);
