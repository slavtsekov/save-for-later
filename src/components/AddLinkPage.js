import React from 'react';

import LinkForm from './LinkForm';

class AddLinkPage extends React.Component {
    onSubmit = (data) => {
        console.log(data);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <h1>Add Link</h1>
                <LinkForm onSubmit={this.onSubmit} />
            </div>
        );
    }
}

export { AddLinkPage as default };
