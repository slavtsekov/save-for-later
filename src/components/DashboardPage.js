import React from 'react';

const DashboardPage = () => (
    <div>
        Dashboard page content.
    </div>
);

export { DashboardPage as default };
