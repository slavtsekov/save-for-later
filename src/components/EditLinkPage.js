import React from 'react';

const EditLinkPage = () => (
    <div>
        Edit Link page content.
    </div>
);

export { EditLinkPage as default };
