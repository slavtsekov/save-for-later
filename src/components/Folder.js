import React from 'react';

const Folder = (props) => (
    <div>
        <p>{props.text}</p>
        <button
            type="button"
            onClick={(ะต) => {
                props.onDelete(props.text);
            }}
        >
            Remove
        </button>
    </div>
);

export { Folder as default };
