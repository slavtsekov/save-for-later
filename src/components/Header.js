import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = (props) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Save For Later</h1>
                </Link>
                <button
                    className="button button--link"
                    type="button"
                    onClick={props.startLogout}
                >
                    Logout
                </button>
            </div>
            <nav>
                <Link to="/dashboard">
                    <h2>Home</h2>
                </Link>
                <Link to="/folders">
                    <h2>Folders</h2>
                </Link>
            </nav>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
