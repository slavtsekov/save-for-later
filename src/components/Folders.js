import React from 'react';

import Folder from './Folder';

const Folders = (props) => {
    const folders = ['Folder 1', 'Folder 2'];
    const handleDeleteFolder = (folderName) => {
        console.log('Removed', folderName);
    };
    return (
        <div>
            <div>
                <h3>Your Folders</h3>
            </div>
            {folders.length === 0 && <p>You have no folders at the moment.</p>}
            {
                folders.map((folder, index) => (
                    <Folder
                        key={folder}
                        text={folder}
                        onDelete={handleDeleteFolder}
                    />
                ))
            }
        </div>
    );
};

export { Folders as default };
