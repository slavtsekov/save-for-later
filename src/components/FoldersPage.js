import React from 'react';
import Folders from './Folders';
import AddFolder from './AddFolder';

const FoldersPage = () => (
    <div>
        <Folders />
        <AddFolder />
    </div>
);

export { FoldersPage as default };
