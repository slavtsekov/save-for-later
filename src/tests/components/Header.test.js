import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

test('should render Header correctly', () => {
    const wrapper = shallow(<Header startLogout={() => {}} />);

    expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
    const startLogoutMock = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogoutMock} />);

    wrapper.find('button').simulate('click');

    expect(startLogoutMock).toHaveBeenCalled();
});
