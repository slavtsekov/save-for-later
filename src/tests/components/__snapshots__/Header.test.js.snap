// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Save For Later
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </div>
    <nav>
      <Link
        replace={false}
        to="/dashboard"
      >
        <h2>
          Home
        </h2>
      </Link>
      <Link
        replace={false}
        to="/folders"
      >
        <h2>
          Folders
        </h2>
      </Link>
    </nav>
  </div>
</header>
`;
